test/threadstest.c

```diff
@@ -741,6 +745,330 @@ static int test_bio_dgram_pair(void)
 }
 #endif
 
+static int test_thread_reported_flags(void)
+{
+    uint32_t flags = OSSL_get_thread_support_flags();
+
+#if !defined(OPENSSL_THREADS)
+    if (!TEST_int_eq(flags, 0))
+        return 0;
+#endif
+
+#if defined(OPENSSL_NO_THREAD_POOL)
+    if (!TEST_int_eq(flags & OSSL_THREAD_SUPPORT_FLAG_THREAD_POOL, 0))
+        return 0;
+#else
+    if (!TEST_int_eq(flags & OSSL_THREAD_SUPPORT_FLAG_THREAD_POOL, 
+                     OSSL_THREAD_SUPPORT_FLAG_THREAD_POOL))
+        return 0;
+#endif
+
+#if defined(OPENSSL_NO_DEFAULT_THREAD_POOL)
+    if (!TEST_int_eq(flags & OSSL_THREAD_SUPPORT_FLAG_DEFAULT_SPAWN, 0))
+        return 0;
+#else
+    if (!TEST_int_eq(flags & OSSL_THREAD_SUPPORT_FLAG_DEFAULT_SPAWN,
+                     OSSL_THREAD_SUPPORT_FLAG_DEFAULT_SPAWN))
+        return 0;
+#endif
+
+    return 1;
+}
+
+#if defined(OPENSSL_THREADS)
+
+# define TEST_THREAD_NATIVE_FN_SET_VALUE 1
+static uint32_t test_thread_native_fn(void *data)
+{
+    uint32_t *ldata = (uint32_t*) data;
+    *ldata = *ldata + 1;
+    return *ldata - 1;
+}
+
+static uint32_t test_thread_noreturn(void *data)
+{
+    CRYPTO_MUTEX *lock = (uint32_t*) data;
+
+    /* lock is assumed to be locked */
+    ossl_crypto_mutex_lock(lock);
+
+    /*
+     * if we ever get here, the test is written wrong; to ensure
+     * that this function is ossl_noreturn, call OPENSSL_die
```

Comment:
The reference to `ossl_noreturn` here is a bit odd since the function is not marked ossl_noreturn. Maybe just s/is ossl_noreturn/does not return/ here.

Reply:
I originally intended to use `ossl_noreturn` (https://github.com/ckalina/openssl/blob/temp3/test/threadstest.c#L788), but windows build complained `D:\a\openssl\openssl\test\threadstest.c(796) : warning C4716: 'test_thread_noreturn': must return a value`. Not having a windows box at hand at the moment to debug this, I simply returned a dummy value (didn't want to get bogged down with this detail). Will s/is ossl_noreturn/does not return/, unless I used `ossl_noreturn` wrongly and you'd prefer me to fix it properly.

Reply:
Comment changed.
```diff
-    /*
-     * if we ever get here, the test is written wrong; to ensure
-     * that this function is ossl_noreturn, call OPENSSL_die
-     */
+    /* unreachable */
```
If that's sufficient, please resolve the conversation :).